### PR TITLE
Set vertical scale per channel

### PR DESCRIPTION
**Pull request by retiutut**
- [x] Refactor TextBox class in Extras.pde to work better than Cp5 Textlabel
- [x] Hide `+yLim` and `-yLim` axis labels when HardwareSettingsController is open
- [x] Hide + and - custom vertical scale buttons when they would start to overlap
- [x] **Convert +yLim/-yLim labels to buttons to make 16ch fit in default window size w/ default layout**
- [x] Hide + and - buttons/labels when autoscale is being used
- [x] **Hard lower limit on 25uV per channel, don't allow users to reverse y-axis or get stuck in autoscale without buttons**
- [x] yAxis label/textfield can accept floats and does `Math.round()`
____
- [x] Update changelog 5.0.1

### CODE REVIEW

**Comment by retiutut** [comment]
Preview using Cyton and Synthetic Data Sources:

Screen Shot 2020-08-08 at 10 55 34 PM
Screen Shot 2020-08-08 at 11 05 32 PM

**Comment by retiutut** [comment]
Screen Shot 2020-08-19 at 10 13 16 PM

**Comment by retiutut** [comment]
Screen Shot 2020-08-19 at 11 22 27 PM

File: OpenBCI_GUI/W_AnalogRead.pde [hunk 1]
```diff
@@ -325,6 +330,38 @@ class AnalogReadBar{
         digitalPin = new TextBox("(D" + digitalPinNum + ")", x+3, y + h + 12);
         digitalPin.textColor = color(bgColor);
         digitalPin.alignH = CENTER;
+        
+
+        /*
```
Comment (Andrey1994) [inline OpenBCI_GUI/W_AnalogRead.pde:335]: can we  remove it?

Comment (retiutut) [reply OpenBCI_GUI/W_AnalogRead.pde:335]: Yes. Improving TextBox class in Extras.pde with methods from Processing can provide reliable justification.

File: OpenBCI_GUI/W_TimeSeries.pde [hunk 2]
```diff
@@ -517,60 +563,48 @@ class ChannelBar{
...
+                fmt = "RAILED - " + fmt;
             } else if (is_railed[channelIndex].is_railed_warn == true) {
-                voltageValue.string = "NEAR RAILED - " + voltageValue.string;
+                fmt = "NEAR RAILED - " + fmt;
             }
         }
+        voltageValue.setText(fmt);
 
         //update the impedance values
         val = data_elec_imp_ohm[channelIndex]/1000;
```
Comment (retiutut) [inline OpenBCI_GUI/W_TimeSeries.pde:602]: I don't really like that these impedance values get divided by 1000 near the code that updates a string just before being drawn.

**Comment by retiutut** [comment]
Last hurdle seems to be when TimeSeries is put in a 1/4 container while at GUI minimum size.
Screen Shot 2020-08-20 at 4 43 21 PM

**Comment by retiutut** [comment]
https://github.com/processing/processing/blob/4cc297c66908899cd29480c202536ecf749854e8/core/src/processing/core/PGraphics.java#L4604

File: OpenBCI_GUI/W_AnalogRead.pde (hunk 1 above)
Comment (retiutut) [reply OpenBCI_GUI/W_AnalogRead.pde]: Cleaned up!

**Comment by retiutut** [comment]
Screen Shot 2020-08-24 at 12 30 28 AM

**Review by Andrey1994 — APPROVED** [review]

**Comment by daniellasry** [comment]
If you're getting pressure from NovaXR peeps to get this change in, don't rush it. Instead, just `git push upstream ts-custom-vertscale` to push you branch to the main repo. This will kick-off a CI build and you can send the link to anyone who wants to test this feature early.

**Comment by retiutut** [comment]
More UI changes coming to this PR...

**Comment by retiutut** [comment]
> I approve of the code, but I saw some issues when testing:
> 
> * The old "Vert Scale" dropdown no longer works. Here's how I think it should work:
>   
>   * When user selects a value in the drop-down, it overrides all channels to use that scale
>   * When a user sets the scale on a specific channel, override the label in the dropdown to read "Custom"
> 
> A usability/design issue we had not anticipated: It's very tedious to adjust the per-channel scale. Imagine we wanted al channels to have a scale of 10,000, and one channel to have a scale of 1,000. It would take _an entire minuted of repeated clicking_ to go from 1,000 to 10,000. A quick fix would be to replace the buttons with a single mini text box where you can type the scale you want per channel.

This is the ultimate "custom vertical scale per channel" hopefully we can combine this with input/mockup using new buttons situated in the center of the channel.

File: OpenBCI_GUI/Extras.pde [hunk 3]
```diff
@@ -258,35 +258,60 @@ class PlotFontInfo {
...
+        x = x1;
+        y = y1;
+        textColor = _textColor;
+        backgroundColor = _backgroundColor;
+        fontSize = _fontSize;
+        font = _font;
+        backgroundEdge_pixels = 1;
+        drawBackground = true;
+        alignH = _alignH;
+        alignV = _alignV;
```
[the remark this answers is not in the document]
Comment (retiutut) [reply OpenBCI_GUI/Extras.pde]: Resolved

```
  TextBox(String s, int x1, int y1) {
        string = s; x = x1; y = y1;
        textColor = color(0);
        backgroundColor = color(255);
        fontSize = 12;
        font = p5;
        backgroundEdge_pixels = 1;
        drawBackground = false;
        alignH = LEFT;
        alignV = BOTTOM;
    }

    TextBox(String s, int x1, int y1, color _textColor, color _backgroundColor, int _alignH, int _alignV) {
        this(s, x1, y1);
        textColor = _textColor;
        backgroundColor = _backgroundColor;
        drawBackground = true;
        alignH = _alignH;
        alignV = _alignV;
    }

    TextBox(String s, int x1, int y1, color _textColor, color _backgroundColor, int _fontSize, PFont _font, int _alignH, int _alignV) {
        this(s, x1, y1, _textColor, _backgroundColor, _alignH, _alignV);
        fontSize = _fontSize;
        font = _font;
    }
```

**Review by daniellasry — APPROVED** [review]
Looks really good, nice work!

**Pull request merged**